.github/templates/linux_ci_workflow.yml.j2

```diff
@@ -493,3 +495,80 @@ jobs:
           docker system prune -af
 
   {%- endif -%}
+
+{%- if enable_update_public_apis %}
+
+  update_public_apis:
+    runs-on: linux.2xlarge
+    needs:
+      - calculate-docker-image
+      - build
+    env:
+      DOCKER_IMAGE: ${{ needs.calculate-docker-image.outputs.docker_image }}
+    steps:
+      - name: Log in to ECR
+        run: |
+          aws ecr get-login --no-include-email --region us-east-1 > /tmp/ecr-login.sh
+          bash /tmp/ecr-login.sh
+          rm /tmp/ecr-login.sh
+      - name: Chown workspace
+        run: |
+          # Ensure the working directory gets chowned back to the current user
+          docker run --rm -v "$(pwd)":/v -w /v "${ALPINE_IMAGE}" chown -R "$(id -u):$(id -g)" .
+      - name: Checkout PyTorch
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # deep clone, to allow sharding to use git rev-list
+          submodules: recursive
+      - name: Pull docker image
+        run: |
+          docker pull "${DOCKER_IMAGE}"
+      - name: Preserve github env variables for use in docker
+        run: |
+          env | grep '^GITHUB' > "/tmp/github_env_${GITHUB_RUN_ID}"
+      - uses: seemethere/download-artifact-s3@0504774707cbc8603d7dca922e8026eb8bf3b47b
+        name: Download PyTorch Build Artifacts
+        with:
+          name: ${{ env.BUILD_ENVIRONMENT }}
+      - name: Unzip artifacts
+        run: |
+          unzip -o artifacts.zip
+      - name: Generate list of public APIs
+        run: |
+          set -ex
+          time docker pull "${DOCKER_IMAGE}" > /dev/null
+          echo "${GITHUB_REF}"
+          docker run \
+            -e BUILD_ENVIRONMENT \
+            -e IN_CI \
+            --security-opt seccomp=unconfined \
+            --name="$GITHUB_SHA" \
+            --tty \
+            --user jenkins \
+            -v "${GITHUB_WORKSPACE}:/var/lib/jenkins/workspace" \
+            -w /var/lib/jenkins/workspace \
+            "${DOCKER_IMAGE}" \
+            bash -c "sudo chown -R jenkins . && pip install dist/*.whl && python scripts/list_apis.py --module torch --public | sort > apis.log"
+      - name: Chown workspace
+        run: |
+          # Ensure the working directory gets chowned back to the current user
+          docker run --rm -v "$(pwd)":/v -w /v "${ALPINE_IMAGE}" chown -R "$(id -u):$(id -g)" .
+      - name: Archive APIs
+        run: |
+          zip apis.zip apis.log
+      - uses: driazati/upload-artifact-s3@50adbe4ef0b6d9221df25c18c5fc528dfcb7c3f8
```

Comment:
it looks like there are three different versions of `upload-artifact-s3` that we use:

- `driazati/upload-artifact-s3@50adbe4ef0b6d9221df25c18c5fc528dfcb7c3f8` (this one), in one place
- `driazati/upload-artifact-s3@21c31d0a7bcb056ca50bd6ce197ba6507c26a1be`, in [one place](https://github.com/pytorch/pytorch/blob/4a544df00dee565e6ccb73e971ff734346b16474/.github/templates/linux_ci_workflow.yml.j2#L484)
- `seemethere/upload-artifact-s3@9d7ceb0ab39c2c88d93ef7792b27425b27d59162`, in [two](https://github.com/pytorch/pytorch/blob/4a544df00dee565e6ccb73e971ff734346b16474/.github/templates/linux_ci_workflow.yml.j2#L188) [places](https://github.com/pytorch/pytorch/blob/4a544df00dee565e6ccb73e971ff734346b16474/.github/templates/windows_ci_workflow.yml.j2#L85)

If it is necessary to use all these different versions, could you add code comment(s) explaining why? And if not, would it be a good idea to consolidate to use a single version?

Reply:
We should definitely consolidate: #61384